idd/Energy+.idd.in

```diff
@@ -34502,6 +34502,4630 @@ ZoneHVAC:EvaporativeCoolerUnit,
+		\note Mode1 - Mode 25 describe active operating modes that provide ventilation, heating, cooling, humidification or dehumidification.
+		\note 26 modes are allowed (including Mode0).
+		\note If the value input to this field is 1, only Mode0 will be considered.
+		\note If this field is blank, the default number of operating modes is 1.
+		\default 1
+	N5,	\field Minimum Time Between Mode Change
+		\type real
+		\minimum 1
+		\note Any mode selected will not operate for less time than the value input in this field. If the value in this field is larger than each timestep, the mode and settings selected in one time step will persist in later time steps until the minimum time between mode change is satisfied. Mode0 does not have a minimum time. If this field is blank, the default minimum time between mode change is 10 minutes.
+		\units minutes
+		\default 10
+	A13,	\field First fuel type
+		\type choice
+		\key Electricity
+		\default Electricity
+		\note Select the fuel type associated with field: "System Electric Power Lookup Table" in each mode. 
+	A14,	\field Second fuel type
+		\type choice
+		\key None
+		\key Electricity
+		\key NaturalGas
+		\key PropaneGas
+		\key FuelOil#1
+		\key FuelOil#2
+		\key Diesel
+		\key Gasoline
+		\key Coal
+		\key OtherFuel1
+		\key OtherFuel2
+		\key Steam
+		\key DistrictHeating
+		\key DistrictCooling
+		\note Select the fuel type associated with field: "System Second Fuel Consumption Lookup Table" in each mode.
+		\note If this field is blank, default second fuel type = None.
+		\default None
+	A15,	\field Third fuel type
+		\type choice
+		\key None
+		\key Electricity
+		\key NaturalGas
+		\key PropaneGas
+		\key FuelOil#1
+		\key FuelOil#2
+		\key Diesel
+		\key Gasoline
+		\key Coal
+		\key OtherFuel1
+		\key OtherFuel2
+		\key Steam
+		\key DistrictHeating
+		\key DistrictCooling
+		\note Select the fuel type associated with field: "System Third Fuel Consumption Lookup Table" in each mode.
+		\note If this field is blank, default third fuel type = None.
+		\default None
+	A16,	\field Objective Function Minimizes
+		\type choice
+		\key Electricity Use
+		\key Second Fuel Use
+		\key Third Fuel Use
+		\key Water Use
+		\note In each time step, controlled variables will be chosen to minimize the selection in this field, subject to constraints.
+		\note If this field is blank, the objective function will minimize electricity use.
+		\default Electricity Use
+	A17,	\field Design Specification Outdoor Air Object Name
+		\type object-list
+		\object-list DesignSpecificationOutdoorAirNames
+		\note Enter the name of a DesignSpecification:OutdoorAir object. Information in that object will be used to compute the minimum outside air flow rate in each time step.
+		\note If this field is blank, the system may still supply outside air, if it is capable as described by lookup tables, when doing so is the most efficient way to satisfy other constraints.
+	A18,	\field Mode0 Name
```

Comment:
Okay re: spaces.

I agree that creating another object class to describe modes would make it easier to adapt the  structure of inputs that define each mode later on, if there were a need to change. However, in such a circumstance, there would be so many other things that need to change (eg: TPEX data entry form, TPEX performance map template, TPEX-BCL translator, update to all BCL archived data) that the changes required to update the IDD and model code would be small in comparison.  

Perhaps more importantly, creating an object class for each mode could create the impression that a user could mix and match modes into a system configuration however they please.  in reality, a group of modes is distinct to a particular product, they cannot be separated.  It is therefore intuitive that all information for a single machine be represented within a single object.